### PR TITLE
Updating search component to use classnames and string templates

### PR DESCRIPTION
**Pull request by jwijay**

### CODE REVIEW

File: src/js/components/Search.js [hunk 1]
```diff
@@ -274,55 +263,60 @@ export default class Search extends Component {
...
-    }
-    if (this.props.large) {
-      classes.push(CLASS_ROOT + "__drop--large");
-    }
+    let classes = classnames (
+      {
+        [`background-color-index-${this.props.dropColorIndex}`]: this.props.dropColorIndex,
+        [`${CLASS_ROOT}__drop`]: true,
+        [`${CLASS_ROOT}__drop--controlled`]: !(this.state.inline),
+        [`${CLASS_ROOT}__drop--inline`]: this.state.inline,
```
Comment (ericsoderberghp) [inline src/js/components/Search.js]: Does the drop need the --inline class?

File: src/js/components/Search.js [hunk 2]
```diff
@@ -333,36 +327,34 @@ export default class Search extends Component {
...
-    }
-    if (this.props.className) {
-      classes.push(this.props.className);
-    }
-    if (this.props.iconAlign) {
-      classes.push(CLASS_ROOT + "--icon-align-" + this.props.iconAlign);
-    }
+    let classes = classnames(
+      CLASS_ROOT,
+      this.props.className,
```
Comment (ericsoderberghp) [inline src/js/components/Search.js]: Can we put this one at the end of the list? It makes it slightly easier to parse the classes in the DOM when they are grouped according to which component sets them.

**Comment by ericsoderberghp** [comment]
Very nice cleanup. Thanks.

File: src/js/components/Search.js [hunk 3]
```diff
@@ -333,39 +326,35 @@ export default class Search extends Component {
...
-      classes.push(this.props.className);
-    }
-    if (this.props.iconAlign) {
-      classes.push(CLASS_ROOT + "--icon-align-" + this.props.iconAlign);
-    }
+    let classes = classnames(
+      CLASS_ROOT,
+      {
+        [`${CLASS_ROOT}--controlled`]: !(this.state.inline),
+        [`${CLASS_ROOT}----fill`]: this.props.fill,
```
Comment (ericsoderberghp) [inline src/js/components/Search.js]: Looks like too many '-' here.

Comment (jwijay) [inline src/js/components/Search.js]: Thanks for catching that. Updated the PR.

**Pull request merged**